src/ecalc/libraries/libecalc/common/libecalc/core/graph_result.py

```diff
@@ -343,3 +388,43 @@ def get_results(self) -> EnergyCalculatorResult:
             emission_results=self.emission_results,
             variables_map=self.variables_map,
         )
+
+    def get_pressures_from_temporal_models(
+        self, energy_usage_model: dict, regularity: dict, pressure_type: CompressorPressureState
+    ) -> Dict[datetime.datetime, TimeSeriesFloat]:
+        """Extract compressor input pressures from temporal models.
+
+        :param energy_usage_model: dictionary of temporal energy models
+        :param regularity: regularity for the actual component
+        :param pressure_type: type of pressure to evaluate (inlet- or outlet pressure)
+        :return: inlet- and outlet input pressure time series
+        """
+
+        pressures = {}
+
+        for time_value, model in energy_usage_model.items():
+            pressure = model.suction_pressure
+            if pressure_type.value == pressure_type.OUTLET_PRESSURE:
+                pressure = model.discharge_pressure
+
+            if pressure is not None:
+                pressure_value = pressure
+
+                if isinstance(pressure, Expression):
+                    pressure_value = list(
+                        pressure.evaluate(self.variables_map, fill_length=len(self.variables_map.time_vector))
+                    )
+
+            else:
+                pressure_value = [math.nan] * len(self.timesteps)
+
+            pressure_time_series = TimeSeriesFloat(
+                timesteps=self.timesteps,
+                values=pressure_value,
+                unit=Unit.BARA,
+                regularity=regularity,
+            )
+
+            pressures[time_value] = pressure_time_series
+
+        return pressures
```

Comment:
Is it necessary to keep the temporal part here? Can't we merge the lists into one?

Reply:
I thought the same, but not sure ...

Reply:
I am looking at it:-)

Reply:
The temporal models are merged into one time series now.